We-the-People-civ4col-mod/Mod
# Bobisback Import Changes

## Description
- Added in a new button on the import screen that allows with one click all demanded goods to be imported into the city.
- Added in a new button on the import screen that allows with one click all goods needed for construction to be imported into the city.
- Added all relevant defines to the global defines file for semi convenient customization. 
- Included a small bug fix in the auto export code that was removing food from imports and exports.


## Known Impacted Areas of Code

This should only effect the import/export popup. This also changes some code related to the auto trader code related to some small bugs I found.

## Additional Notes
* We have to add import limits on everything so it plays nice with the auto export feature.
* I did fix a bug in the auto export that was removing food as a import or export no matter what.
* This code is not great but please keep in mind I copied to be the same as the code that is already there. This way when it is refactored it can all be done together. This should keep things cleaner for the future. 
* I recommend who ever merges this PR squashes my commits, since I do have a bunch of commits in there. 

## Details for Code Reviewer Running Code

Steps to test for importing demanded goods

1. Run game
2. Enter a city that has demanded goods
3. Optionally reset imports for the city via the imports popup
4. Use the import all demanded goods button
5. You should observe that all goods that are demanded are set to a 5 turns worth of goods for the maintain level and 8 turns worth of goods for the import limit level. 
6. Done! Report any issues or questions here and has much success!!

Steps to test for importing Construction goods

1. Run game
2. Enter any city
3. Optionally reset imports for the city via the imports popup
4. Use the import construction goods button
5. You should observe that construction goods are set to maintain and limit values that are in the global defines
 (100 for lumber, clay, and stone, 150 for tools) Again we have to set limit values otherwise it will break auto export.
6. Done! Report any issues or questions here and has much success!!

Steps to test for auto export bug

1. Run game
2. Enter any city
3. Add a food import or export
4. Use the auto export all goods button
5. You should observe that the food import/export should be unchanged
6. Done! Report any issues or questions here and has much success!!

### Author's Checklist
- [x] Did you run the code and verify that it works?
- [x] Is the code refined/iterated/beautiful?
- [x] Does your stuff deallocate well (ie was things cleaned up after being used)?
- [x] Are warnings in classes you are using cleaned up or suppressed?
- [x] Did you think about networking and not creating desyncs?
- [x] Privacy: Do you prevent any sensitive user information from being exposed?
- [x] Security: Have you considered and addressed any potential security concerns?
- [x] Did you leave relevant but not overstated comments in your code?

### Other Reviewers

- [x] Did you run the code? Does it work? Add screenshots/videos as needed for failed items.
- [x] Is the code as abstracted, composable, and isolated as possible?
- [x] Privacy: Is any user information being exposed that shouldn’t be?
- [x] Security: Does this have any security concerns?
- [x] Are your comments to the author positive?
